service/history/hsm/tree.go

```diff
@@ -144,13 +177,19 @@ func NewRoot(registry *Registry, t string, data any, children map[string]*persis
 			InitialVersionedTransition:    &persistencespb.VersionedTransition{},
 			LastUpdateVersionedTransition: &persistencespb.VersionedTransition{},
 			TransitionCount:               0,
+			Deleted:                       false,
```

Comment:
I think it's good that you've added the deleted flag to prevent deleted nodes from being used but the flag should be on the cache, we won't persist deleted nodes. You should enforce no further transitions or deletions on a deleted node.

Reply:
Done